src/backend/distributed/utils/maintenanced.c

```diff
@@ -298,6 +298,16 @@ CitusMaintenanceDaemonMain(Datum main_arg)
 		double timeout = 10000.0; /* use this if the deadlock detection is disabled */
 		bool foundDeadlock = false;
 
+		LWLockAcquire(&MaintenanceDaemonControl->lock, LW_EXCLUSIVE);
+		pid_t dbDataWorkerPid = myDbData->workerPid;
+		LWLockRelease(&MaintenanceDaemonControl->lock);
+
+		if (dbDataWorkerPid != MyProcPid)
+		{
+			/* if multiple daemons are alive, only keep the one in the hash table */
+			proc_exit(1);
```

Comment:
Maybe `proc_exit(0)` is better if restart isn't wanted. Might also be good to define a constant to clarify what proc_exit(1) means for maintenanced